sdk/search/Azure.Search/src/SearchFilter.cs

```diff
@@ -0,0 +1,113 @@
+        /// </param>
+        /// <returns>A valid OData filter expression.</returns>
+        public static string Create(FormattableString filter, IFormatProvider formatProvider)
+        {
+            if (filter == null) { return null; }
+            formatProvider ??= CultureInfo.InvariantCulture;
+
+            string[] args = new string[filter.ArgumentCount];
+            for (int i = 0; i < filter.ArgumentCount; i++)
+            {
+                args[i] = filter.GetArgument(i) switch
+                {
+                    // Null
+                    null => "null",
+
+                    // Boolean
+                    bool x => x.ToString(formatProvider).ToLowerInvariant(),
+
+                    // Numeric
+                    sbyte x => x.ToString(formatProvider),
+                    byte x => x.ToString(formatProvider),
+                    short x => x.ToString(formatProvider),
+                    ushort x => x.ToString(formatProvider),
+                    int x => x.ToString(formatProvider),
+                    uint x => x.ToString(formatProvider),
+                    long x => x.ToString(formatProvider),
+                    ulong x => x.ToString(formatProvider),
+                    decimal x => x.ToString(formatProvider),
+
+                    // Floating point
+                    float x => JsonExtensions.Float(x, formatProvider),
+                    double x => JsonExtensions.Double(x, formatProvider),
+
+                    // Dates as 8601 with a time zone
+                    DateTimeOffset x => JsonExtensions.Date(x, formatProvider),
+                    DateTime x => JsonExtensions.Date(x, formatProvider),
+
+                    // TODO: XXXXX - Unify on an Azure.Core spatial type
+
+                    // Text
+                    string x => Quote(x),
+                    char x => Quote(x.ToString(formatProvider)),
+                    StringBuilder x => Quote(x.ToString()),
+
+                    // Everything else
+                    object x => throw new ArgumentException(
+                        $"Unable to convert argument {i} from type {x.GetType()} to an OData literal.")
+                };
+            }
+            string text = string.Format(formatProvider, filter.Format, args);
+            return text;
+        }
+
+        /// <summary>
+        /// Quote and escape OData strings.
+        /// </summary>
+        /// <param name="text">The text to quote.</param>
+        /// <returns>The quoted text.</returns>
+        private static string Quote(string text)
+        {
+            if (text == null) { return "null"; }
+
+            // Optimistically allocate an extra 5% for escapes
+            StringBuilder builder = new StringBuilder(2 + (int)(text.Length * 1.05));
+            builder.Append("'");
+            foreach (char ch in text)
+            {
+                builder.Append(ch);
+                if (ch == '\'')
```

Comment:
Do we need to worry about double quotes as well, or is that not possible in an OData filter?

Reply:
Only single quotes are allowed as string delimiters in OData expressions.